### PR TITLE
Add kb1200 soc

### PR DESCRIPTION
**Pull request by ene-steven**
Add ENE KB1200 SOC support.

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]
> Merge branch 'main' into add_kb1200_soc

Please squash changes and do not add merge commits

File: dts/arm/ene_kb1200/kb1200_dtsi.h [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+#define FSMBM1_BASE         (0x40341000UL)
+#define FSMBM2_BASE         (0x40342000UL)
+#define FSMBM3_BASE         (0x40343000UL)
+#define FSMBM4_BASE         (0x40344000UL)
+#define FSMBM5_BASE         (0x40345000UL)
+#define FSMBM6_BASE         (0x40346000UL)
+#define FSMBM7_BASE         (0x40347000UL)
+#define FSMBM8_BASE         (0x40348000UL)
+#define FSMBM9_BASE         (0x40349000UL)
+#define I2CS0_BASE          (0x40350000UL)
```
Comment (nordicjm) [inline dts/arm/ene_kb1200/kb1200_dtsi.h:108]: Why is this not in a separate HAL?

File: drivers/serial/uart_kb1200.c [hunk 2]
```diff
@@ -0,0 +1,696 @@
...
+	uint16_t tx;
+	uint16_t rx;
+	uint8_t  pintype;
+};
+
+static const struct kb1200_uart_pins uart_pin_cfg[] = {
+	{ SER1_TX_GPIO_Num, SER1_RX_GPIO_Num , PINMUX_FUNC_B},   /* SER1 */
+	{ SER2_TX_GPIO_Num, SER2_RX_GPIO_Num , PINMUX_FUNC_C},   /* SER2 */
+	{ SER3_TX_GPIO_Num, SER3_RX_GPIO_Num , PINMUX_FUNC_B},   /* SER3 */
+};
```
Comment (gmarull) [inline drivers/serial/uart_kb1200.c:46]: please implement a proper pinctrl driver

**Review by gmarull — CHANGES_REQUESTED** [review]
please, format code according to project rules, also use devicetree for any hardware description

**Comment by martinjaeger** [comment]
Hello @ene-steven,

As mentioned before in #63028, please read the [Zephyr contribution guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html) carefully and follow them.

And please stop opening new PRs, but fix the existing one instead.

If you have questions, feel free to [join the Discord](https://chat.zephyrproject.org/) and ask the community.

File: drivers/gpio/CMakeLists.txt [hunk 3]
```diff
@@ -83,7 +83,7 @@ zephyr_library_sources_ifdef(CONFIG_GPIO_NUMAKER    gpio_numaker.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_EFINIX_SAPPHIRE    gpio_efinix_sapphire.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_DAVINCI    gpio_davinci.c)
 zephyr_library_sources_ifdef(CONFIG_GPIO_SEDI    gpio_sedi.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_ALTERA_PIO gpio_altera_pio.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO_KB1200     gpio_kb1200.c)
```
Comment (martinjaeger) [inline drivers/gpio/CMakeLists.txt:86]: Don't delete the `altera_pio`.

**Review by martinjaeger — CHANGES_REQUESTED** [review]

**Pull request closed without merging**